pkg/kubelet/pod/pod_manager_test.go

```diff
@@ -17,151 +17,207 @@ limitations under the License.
+		{
+			name:                             "Remove static pod",
+			podList:                          []*v1.Pod{mirrorPod, staticPod, normalPod},
+			needToRemovePod:                  staticPod,
+			expectPods:                       []*v1.Pod{normalPod},
+			expectMirrorPods:                 []*v1.Pod{mirrorPod},
+			expectOrphanedMirrorPodFullnames: []string{"mirror-static-pod-name_default"},
 		},
-	}
-
-	expectedPods := []*v1.Pod{
 		{
-			ObjectMeta: metav1.ObjectMeta{
-				UID:         types.UID("extra-pod-uid"),
-				Name:        "extra-pod-name",
-				Namespace:   metav1.NamespaceDefault,
-				Annotations: map[string]string{kubetypes.ConfigSourceAnnotationKey: "api"},
-			},
+			name:             "Remove normal pod",
+			podList:          []*v1.Pod{mirrorPod, staticPod, normalPod},
+			needToRemovePod:  normalPod,
+			expectPods:       []*v1.Pod{staticPod},
+			expectMirrorPods: []*v1.Pod{mirrorPod},
 		},
-		staticPod,
-	}
-	updates := append(expectedPods, mirrorPod)
-	podManager, _ := newTestManager()
-	podManager.SetPods(updates)
-
-	podManager.RemovePod(staticPod)
-
-	actualPods := podManager.GetPods()
-	if len(actualPods) == len(expectedPods) {
-		t.Fatalf("Run RemovePod() error, expected %d pods, got %d pods; ", len(expectedPods)-1, len(actualPods))
-	}
-
-	_, _, orphanedMirrorPodNames := podManager.GetPodsAndMirrorPods()
-	expectedOrphanedMirrorPodNameNum := 1
-	if len(orphanedMirrorPodNames) != expectedOrphanedMirrorPodNameNum {
-		t.Fatalf("Run getOrphanedMirrorPodNames() error, expected %d orphaned mirror pods, got %d orphaned mirror pods; ", expectedOrphanedMirrorPodNameNum, len(orphanedMirrorPodNames))
 	}
 
-	expectedOrphanedMirrorPodName := mirrorPod.Name + "_" + mirrorPod.Namespace
-	if orphanedMirrorPodNames[0] != expectedOrphanedMirrorPodName {
-		t.Fatalf("Run getOrphanedMirrorPodNames() error, expected orphaned mirror pod name : %s, got orphaned mirror pod name %s; ", expectedOrphanedMirrorPodName, orphanedMirrorPodNames[0])
+	for _, test := range testCase {
+		t.Run(test.name, func(t *testing.T) {
+			podManager := NewBasicPodManager()
+			podManager.SetPods(test.podList)
+			podManager.RemovePod(test.needToRemovePod)
+			actualPods1 := podManager.GetPods()
+			actualPods2, actualMirrorPods, orphanedMirrorPodFullnames := podManager.GetPodsAndMirrorPods()
+			// Check if the actual pods and mirror pods match the expected ones.
+			assert.Empty(t, cmp.Diff(actualPods1, actualPods2, sortOpt), "actualPods by GetPods() and GetPodsAndMirrorPods() differ")
+			assert.Empty(t, cmp.Diff(actualPods1, test.expectPods, sortOpt), "actualPods and expectPods differ")
+			assert.Empty(t, cmp.Diff(actualMirrorPods, test.expectMirrorPods, sortOpt), "actualMirrorPods and expectMirrorPods differ")
+			assert.Empty(t, cmp.Diff(orphanedMirrorPodFullnames, test.expectOrphanedMirrorPodFullnames, sortOpt), "orphanedMirrorPodFullnames and expectOrphanedMirrorPodFullnames differ")
```

Comment:
How `sortOpt` is used here? `orphanedMirrorPodFullnames` is a `[]string` and doesn't have a `Name` field.